### ISSUE TITLE
LUA errors

### ISSUE THREAD

**Issue by Ciolouse** [Caerdon/CaerdonWardrobe#1]
Greetings,
Thanks for this great addon !
I do have some LUA errors:

```
1x CaerdonWardrobe\Core.lua:1725: invalid key to 'next'
[C]: in function `error'
CaerdonWardrobe\Core.lua:1725: in function <CaerdonWardrobe\Core.lua:1720>

Locals:
(*temporary) = "invalid key to 'next'"
```

```
1x [ADDON_ACTION_BLOCKED] L'AddOn 'CaerdonWardrobe' a tenté d'appeler la fonction protégée 'CompactPartyFrameMember1:Show()'.
!BugGrabber\BugGrabber.lua:519: in function <!BugGrabber\BugGrabber.lua:519>
[C]: in function `Show'
FrameXML\CompactUnitFrame.lua:356: in function `CompactUnitFrame_UpdateVisible'
FrameXML\CompactUnitFrame.lua:289: in function `CompactUnitFrame_UpdateAll'
FrameXML\CompactUnitFrame.lua:118: in function <FrameXML\CompactUnitFrame.lua:54>

Locals:
InCombatSkipped
```

I'm using version 2.1.5.
Thanks !

**Comment by Caerdon** [comment]
Hmm... strange.  Can you confirm if it happens on older versions of the add-on?  The error isn't giving me much to go on as I didn't make any changes that should have directly impacted that section of the code, so you may have to see what you can do to narrow down the problem.  Does it happen on all toons?  Is it tied to an action you perform (opening bags or bank)?  If so, can you move items out of the thing causing a problem and figure out if it's tied to a specific item?

**Comment by Caerdon** [comment]
The second error is often symptomatic of another add-on problem.  You may want to disable all other addons while you work to identify the cause of the first error.  If both suddenly go away, you'll probably have to dig through your addons and figure out which one is causing conflicts.

**Comment by Ciolouse** [comment]
Greetings,
After disable all addons, I bought some amors in the HV and get this error : 

```
2x CaerdonWardrobe\Core.lua:1827: invalid key to 'next'
[C]: in function `error'
CaerdonWardrobe\Core.lua:1827: in function <CaerdonWardrobe\Core.lua:1822>

Locals:
(*temporary) = "invalid key to 'next'"

Using v2.2.1.
```

**Comment by Caerdon** [comment]
Thanks.  I was finally able to reproduce this, and it should be fixed in release 2.2.2.
